source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -145,14 +145,18 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
             argData.getFlagArgument(kOgsXmlFlag, 0, ogsXmlFileName);
         }
 
+        mx::DocumentPtr document = MaterialXMaya::loadDocument(
+            documentFilePath.asChar(), Plugin::instance().getLibrarySearchPath()
+        );
+
         std::unique_ptr<MaterialXData> materialXData{
-            new MaterialXData(materialXDocument.asChar(),
-                              elementPath.asChar(),
+            new MaterialXData(document,
+                              xmlElementPath.asChar(),
```

Comment:
This is not an xml element path it's an element path in the document.

Reply:
Well, it maps to an XML element in the serialized document but I agree that it's an implementation detail that we probably want to hide.